### ISSUE TITLE
Semantic change for same template fragments in 3.x 

### ISSUE THREAD

**Issue by wienczny** [thymeleaf/thymeleaf#428]
The semantic of same template fragements changed between 2.x and 3.x. 
In 2.x I'm using this feature for page templates that are wrapped around a content template. In 3.x this does not work any more:

In the page template is a "fragement"-declaration at the html element and a "replace" using the fragment "content" that is contained in "this".
In the content page is a "replace" at the html element invoking the page template and a declaration of the "content"-fragment.
It seems the "content" fragment is not contained in "this" in 3.x like it was in 2.x. 
Is this an intended change? Is there an alternative besides the thymeleaf-layout-dialect?

Page template:

```
<!DOCTYPE html>
<html xmlns:th="http://www.thymeleaf.org" th:fragment="page">
<head>
</head>
<body>
<span>Before</span>
<div th:replace="this::content">Content</div>
<span>After</span>
</body>
</html>
```

Invoked content template:

```
<!DOCTYPE html>
<html xmlns:th="http://www.thymeleaf.org" th:replace="template :: page">
<head>
    <title></title>
</head>
<body>
<div th:fragment="content">Expected Content</div>
</body>
</html>
```

In 3.x it's output is 

```
<!DOCTYPE html>
<html>
<head>
</head>
<body>
<span>Before</span>

<span>After</span>
</body>
</html>
```

In 2.x the output would have been

```
<!DOCTYPE html>
<html>
<head>
</head>
<body>
<span>Before</span>
<div>Expected Content</div>
<span>After</span>
</body>
</html>
```

I've created a small template project showing the issue https://github.com/wienczny/thymeleaf-example-same-template-fragement

**Comment by seanhinkley** [comment]
+1

**Comment by danielfernandez** [comment]
Fixed by means of #434

**Comment by danielfernandez** [comment]
Already in `3.0.0-SNAPSHOT`
